clippy_lints/src/loops.rs

```diff
@@ -578,48 +578,55 @@ fn check_for_loop_reverse_range(cx: &LateContext, arg: &Expr, expr: &Expr) {
     }
 }
 
+fn lint_iter_method(cx: &LateContext, args: &[Expr], arg: &Expr, method_name: &str) {
+    let object = snippet(cx, args[0].span, "_");
+    let muta = if method_name == "iter_mut" {
+        "mut "
+    } else {
+        ""
+    };
+    span_lint_and_sugg(cx,
+                       EXPLICIT_ITER_LOOP,
+                       arg.span,
+                       "it is more idiomatic to loop over references to containers instead of using explicit \
+                        iteration methods",
+                       "to write this more concisely, try looping over",
+                       format!("&{}{}", muta, object))
```

Comment:
But the tests are not ui-tests. The suggestion was included after “over”, now that “try looping over” will be followed by the whole line of the replacement.
We'll have something like
```rust
help: try looping over
  |     for foo in &bar {
  | }
```